app/jobs/base_job.rb

```diff
@@ -8,36 +8,45 @@ class BaseJob < ApplicationJob
   STARTING_PATH = File.realpath(FileUtils.pwd)
   include NotifierConfig
 
-  if ENV['ECS'] != 'true'
-    # When called through Active::Job, uses this hook
-    before_perform do |job|
-      unless running_in_correct_path?
+  attr_accessor :start_time
 
-        msg = "Started dir is `#{STARTING_PATH}`\nCurrent dir is `#{expected_path}`\nExiting in order to let systemd restart me in the correct directory. Active::Job"
-        notify_on_restart(msg)
+  if ENV['EKS'] == 'true'
+    # I can't get this to work correctly from the delayed_job initializer
+    Rails.logger.info 'Bootstrapping prometheus metrics'
+    DjMetrics.instance.register_metrics_for_delayed_job_worker!
 
-        unlock_job!(job.job_id) if job.respond_to? :job_id
+    rescue_from StandardError do |err|
+      DjMetrics.instance.dj_job_status_total_metric.increment(labels: { queue: queue_name, priority: priority, status: 'failure', job_name: self.class.name })
+      raise err
+    end
 
-        # Exit, ignoring signal handlers which would prevent the process from dying
-        exit!(0)
-      end
+    # When called through Active::Job, uses this hook
+    before_perform do |job|
+      before_handler(job)
+    end
+
+    after_perform do |job|
+      after_handler(job)
     end
-  end
 
-  if ENV['ECS'] != 'true'
     # When called through Delayed::Job, uses this hook
     def before(job)
-      return if running_in_correct_path?
-
-      job = self unless job.respond_to? :locked_by
+      before_handler(job)
+    end
 
-      msg = "Started dir is `#{STARTING_PATH}`\nCurrent dir is `#{expected_path}`\nExiting in order to let systemd restart me in the correct directory. Delayed::Job"
-      notify_on_restart(msg)
+    def after(job)
+      after_handler(job)
+    end
 
-      unlock_job!(job.id)
+    def before_handler(job)
+      self.start_time = Time.now
```

Comment:
It's probably fine since you're using it all in the same file and just getting the time diff, but we tend to prefer `Time.current` over `Time.now` to incorporate the timezone.